teachertool/src/types/index.ts

```diff
@@ -44,3 +46,11 @@ export type CarouselCardSet = {
 };
 
 export type RequestStatus = "init" | "loading" | "error" | "success";
+
+export type ConfirmationModalProps = {
```

Comment:
The inclusion of `Props` in the name tells me this configures a React component. Props is uncommon for non-React configuration. I suggest using something else here. Suggestions:
`ConfirmationModalOptions`
`ConfirmationModalParams`